static/app/views/replays/detail/network/networkDetailsContent.tsx

```diff
@@ -51,42 +98,111 @@ function DetailsTab({item, startTimestampMs}: TabProps) {
       <SectionItem title={t('General')}>
         {keyValueTablOrNotFound(data, t('Missing request details'))}
       </SectionItem>
-      <SectionItem title={t('Request Headers')}>
-        {keyValueTablOrNotFound(item.data.request?.headers, t('Headers not captured'))}
-      </SectionItem>
-      <SectionItem title={t('Response Headers')}>
-        {keyValueTablOrNotFound(item.data.request?.headers, t('Headers not captured'))}
-      </SectionItem>
+      {content}
     </SectionList>
   );
 }
 
-function RequestTab({item}: TabProps) {
+function RequestTab({item, projectId}: TabProps) {
+  const organization = useOrganization();
+  const sdkNeedsUpdate = useSDKNeedsUpdate({
+    minVersion: MIN_REPLAY_NETWORK_BODIES_SDK,
+    organization,
+    projectId,
+  });
+
+  const warnings = item.data.request?._meta?.warnings as undefined | string[];
+  const requestbody = item.data.request?.body;
+
+  const isUrlSkipped = warnings?.includes('URL_SKIPPED');
+  const isBodySkipped = warnings?.includes('BODY_SKIPPED');
+  const showSetup = sdkNeedsUpdate || isUrlSkipped || isBodySkipped;
+
+  const content = !isSupportedOp(item) ? (
+    <SectionItem title={t('Request Payload')}>
+      <NotFoundText>
+        {t('Request Bodies are only supported on fetch and xhr requests')}
+      </NotFoundText>
+    </SectionItem>
+  ) : showSetup ? (
+    <SectionItem title={t('Request Payload')}>
+      <NetworkDetailsSetup
+        minSDKVersion={MIN_REPLAY_NETWORK_BODIES_SDK}
+        showSnippet="bodies"
+        showSDKUpgrade={sdkNeedsUpdate}
+        url={item.description || ''}
+      />
+    </SectionItem>
+  ) : (
+    <SectionItem title={t('Request Payload')}>
+      <WarningMessage warnings={warnings} />
+      {objectInspectorOrNotFound(requestbody, t('Request Body not found'))}
```

Comment:
I agree. Had it printing `undefined` originally, but the figma shows a message instead

using a `/health/` response as an example, chrome shows "Nothing to preview" instead of undefined. Maybe the check should be more like `if "response" in data then objectInspector() else "Body not found"`